domain/src/SharePooling/Services/QuantityAdjuster.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Domain\SharePooling\Services;
+
+use Domain\SharePooling\Services\Exceptions\QuantityAdjusterException;
+use Domain\SharePooling\ValueObjects\QuantityBreakdown;
+use Domain\SharePooling\ValueObjects\SharePoolingTokenAcquisition;
+use Domain\SharePooling\ValueObjects\SharePoolingTokenAcquisitions;
+use Domain\SharePooling\ValueObjects\SharePoolingTokenDisposal;
+use Domain\SharePooling\ValueObjects\SharePoolingTransactions;
+
+final class QuantityAdjuster
```

Comment:
ℹ️ This service restores the quantities from acquisitions that were previously matched with a disposal that is now being reverted